paddle/fluid/operators/sum_op.cc

```diff
@@ -141,6 +133,26 @@ class SumOp : public framework::OperatorWithKernel {
       PADDLE_ENFORCE_NE(dtype, -1,
                         "Sum operator should have at least one tensor");
 
+#ifdef PADDLE_WITH_MKLDNN
+      if (library == framework::LibraryType::kPlain &&
+          platform::CanMKLDNNBeUsed(ctx) &&
+          static_cast<framework::proto::VarType::Type>(dtype) ==
+              framework::proto::VarType::FP32 &&
+          ctx.OutputVar("Out")->IsType<framework::LoDTensor>()) {
+        bool can_use_mkldnn_sum = true;
+        for (size_t idx = 0; idx < x_vars.size(); ++idx) {
+          if (!x_vars[idx]->IsType<LoDTensor>()) {
+            can_use_mkldnn_sum = false;
+          }
+        }
+        if (can_use_mkldnn_sum) {
```

Comment:
The `can_use_mkldnn_sum` value obtained using the above `for` loop could be calculated in a little bit simpler way, using a condition like this:
```
if (std::all_of(x_vars.begin(), x_vars.end(), [](const Variable* v){return v->IsType<LoDTensor>();})) {
```
Otherwise the `break;` instruction should be used after `can_us_mkldnn_sum = false;` instruction, to exit the loop when the result is already known.

Reply:
Thank you, you are right. I changed to all_of version now.